src/core/Song.cpp

```diff
@@ -813,6 +813,83 @@ void Song::addPatternTrack()
 	Engine::patternStore()->setCurrentPattern(dynamic_cast<PatternTrack*>(t)->patternIndex());
 }
 
+/* @brief Converts all PatternTracks to regular tracks on the Song Editor
+ * @param bool singlePattern Indicates whether we are converting a single
+ * PatternTrack or all of the Song PatternTracks.
+ * @param int pattern If single Pattern is true, indicates the index of the
+ * PatternTrack being converted.
+ */
+void Song::convertPatterntoSE(bool singlePattern /* = false*/, int pattern /* = 0*/)
+{
+	// Get all tracks from the Pattern editor
+	PatternStore* patternStore = Engine::patternStore();
+	TrackList tl = patternStore->tracks();
+
+	// For each track, go through all clips and copy them in every place
+	// the corresponding PatternTrack has PatternClips in.
+	for
+	(
+		TrackList::iterator it = tl.begin();
+		it != tl.end();
+		++it
+	)
```

Comment:
```suggestion
	for (Track* track : tl)
```
`TrackList` is a type alias for `std::vector<Track*>`, so you can use a range-based for loop here.

Each instance of `(*it)->` below would then need to be changed to `track->`.